src/adm.rs

```diff
@@ -252,6 +257,18 @@ pub struct AdmTile {
     pub position: Option<u8>,
 }
 
+impl AdmTile {
```

Comment:
FYI - this doesn't include `id` (or `position` but that's not very relevant) -- `#[derive(Hash)]` would have included them all and gave you the hash impl for free (only difference is the caller needs to create a DefaultHasher itself to pass to every `hash` method call)